### ISSUE TITLE
Update editable label with new structure and design

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly#4469]
After testing in react, I think it makes the most sense to update the structure of editable label to use more semantic elements (toggling between button and input). However, the changing structure means we also need to update the styling. 

[React PR](https://github.com/patternfly/patternfly-react/pull/6472)

[Design issue](https://github.com/patternfly/patternfly-design/issues/1073) for upcoming updates

**Comment by mcarrano** [comment]
@srambach regarding design changes, this is the direction that we are headed:

<img width="279" alt="Screen Shot 2021-11-17 at 2 27 40 PM" src="https://user-images.githubusercontent.com/6619860/142269949-34e9f9a3-8912-4e8e-99be-df454b49d31c.png">

For editable lables, we want to add a dotted underline below the text similar to what we do in definition lists. Let me know if you have questions or need a mre detailed visual spec.

**Comment by srambach** [comment]
@mcarrano What happens when the label becomes active? Does the underline go away, or change? 
Two other things I notice; the offset from the text is smaller in the image above compared to the description list setting. And the underline does not extend all the way to the X; it is only under the text. Is that ok?

![image](https://user-images.githubusercontent.com/19825616/143948562-5d66c340-ac8d-4f47-bd95-12774c80118a.png)

**Comment by mcarrano** [comment]
@srambach I think that the spacing you have is fine. I agree that it makes sense to be consistent with what we did for Description list. I also agree that the underline should not extend all the way to the X. Good question about what happens when editing is active. I hadn't expected the underline to remain after entering edit mode, but maybe it should. What do you think @mceledonia ?

**Comment by mceledonia** [comment]
@gonoraj Did you have any design proposals/mocks for the active state here?

**Comment by gonoraj** [comment]
@mceledonia @mcarrano  I had thought the active state would stay the same, with the blue box appearing for the browser's focus state, but I had made possible options incorporating the label while editing as well. I'm not sure what your thoughts are on this or if it's feasible (I also just added this to [this doc](https://docs.google.com/document/d/1ReoW10DzrrWxdAnzwuXbeTMdGXpFfiIAE9JHu-43o4U/edit?usp=sharing) if you prefer to discuss there). I also agree that I didn't expect the underline to remain after entering edit mode.
![active-state](https://user-images.githubusercontent.com/33614237/144121815-ef00dc8b-ef8c-4f04-89a3-12044795047d.jpg)
